### PR TITLE
docs: explain the purpose of order-only deps for generated files [skip ci]

### PR DESCRIPTION
**Pull request by marc-h38**
The documentation of "order-only" dependencies is limited and their
various purposes are especially not clear. See issue #6391 for a recent
example, search the internet for many more. So mention the particular
purpose here while making the documentation barely longer.

### CODE REVIEW

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -547,11 +547,12 @@ libraries (`.so`, `.dll`, etc) will be linked.
...
-all generated input files, including unknown files. For all input
-files (generated and non-generated), Meson uses the [dependency
-file](https://ninja-build.org/manual.html#ref_headers) generated by
-your compiler to determine when to rebuild sources. The behavior is
-similar for other backends.
+all generated input files, including unknown files. This is needed
+to bootstrap the generation of the real dependencies in the
+[depfile](https://ninja-build.org/manual.html#ref_headers)
+generated by your compiler to determine when to rebuild sources.
+Meson relies on this dependency file for all input files, generated
```
Comment (dcbaker) [inline docs/markdown/Reference-manual.md]: I know this that the original says this, but I 'm pretty sure that `meson` is supposed to be `ninja`

Comment (marc-h38) [reply docs/markdown/Reference-manual.md]: I meant "meson _indirectly_ relies on..." based on the last sentence which is: "The behavior is similar for other backends."

I can either add the missing "indirectly" or change to ninja, whatever works.

Comment (dcbaker) [reply docs/markdown/Reference-manual.md]: I thinks `meson` ... `other backends`. is confusing, as meson is not a backend, ninja is. I'd be just as happy to drop the `other backends part`

**Comment by marc-h38** [comment]
Also submitted similar https://github.com/ninja-build/ninja/pull/1725/files

File: docs/markdown/Reference-manual.md (hunk 1 above)
Comment (marc-h38) [reply docs/markdown/Reference-manual.md]: I (mis?)understand the "similar behavior for other **backends**" at the end of this paragraph to be referring to: "For the Ninja **backend**, Meson will create a build-time order-only dependency..." at the top of this paragraph.

On the other hand, you seem to (mis?)understanding this "similar behavior for other backends" to be referring to the "Meson relies on..." sentence immediately before that I'm modifying, correct?

My modification admittedly creates a little more distance between the top and end of this paragraph.

The git log traces this paragraph back to... a mass import from the wiki in 2017 :-(

Comment (jpakkane) [reply docs/markdown/Reference-manual.md]: This was most likely originally written by me and it was wrong, it should have said Ninja.

**Comment by marc-h38** [comment]
> it should have said Ninja.

Done. Re-opening to retry CI.

**Pull request merged**